app/Controllers/Http/GetsController.ts

```diff
@@ -0,0 +1,25 @@
+import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import Profile from 'App/Models/Profile'
+import User from 'App/Models/User'
+
+export default class PostsController {
+  public async getAllUsers() {
+    return User.all()
+  }
+
+  public async getUserProfile({ auth }: HttpContextContract) {
+    try {
+      const user = auth.use('api').user!
+      const profile = await Profile.findByOrFail('userId', user.id)
```

Comment:
Instead of querying separately for user profile. You can use `preload` method on user object.

For eg:

```ts
const user = auth.use('api')?.user
await user.preload('profile');
```

Reply:
Done